connectors/google/google-sheets/src/main/java/io/camunda/connector/gsheets/GoogleSheetsFunction.java

```diff
@@ -6,19 +6,40 @@
  */
 package io.camunda.connector.gsheets;
 
+import com.google.gson.Gson;
 import io.camunda.connector.api.annotation.OutboundConnector;
 import io.camunda.connector.api.outbound.OutboundConnectorContext;
 import io.camunda.connector.api.outbound.OutboundConnectorFunction;
+import io.camunda.connector.gsheets.model.request.GoogleSheetsRequest;
+import io.camunda.connector.gsheets.operation.GoogleSheetOperation;
+import io.camunda.connector.gsheets.supplier.GsonSheetsComponentSupplier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 @OutboundConnector(
     name = "GOOGLESHEETS",
     inputVariables = {"authentication", "operation"},
     type = "io.camunda:google-sheets:1")
 public class GoogleSheetsFunction implements OutboundConnectorFunction {
 
+  private static final Logger LOGGER = LoggerFactory.getLogger(GoogleSheetsFunction.class);
+  private final Gson gson;
+  private final GoogleSheetsOperationFactory operationFactory;
+
+  public GoogleSheetsFunction() {
+    gson = GsonSheetsComponentSupplier.gsonInstance();
+    operationFactory = GoogleSheetsOperationFactory.getInstance();
+  }
+
   @Override
-  public Object execute(OutboundConnectorContext context) throws Exception {
+  public Object execute(OutboundConnectorContext context) {
+    String variables = context.getVariables();
+    final GoogleSheetsRequest request = gson.fromJson(variables, GoogleSheetsRequest.class);
+    context.validate(request);
+    context.replaceSecrets(request); // TODO add secrets to request
```

Comment:
Still `TODO` or is it done?

Reply:
I've pushed already commits with it. So it's done